### PR TITLE
core: workaround ceph-volume's wal/db VG spanning issue

### PR DESCRIPTION
**Pull request by jschmid1**
* Work around c-v's issue described here: http://tracker.ceph.com/issues/37502
  We now create separate calls for each wal/db.

Bonus:
* Improve access time of disks(dg).c_v_commands (and successive .deploy) by 30%
  by reducing the amount of dynamic data/db/wal_devices invocations.

* Add log.debug statements in _filter_devices

* Add mechanism to prevent the listing of the same devices under different
  columns (wal/db/data).
  This is achieved by tracking the assigned devices internally.
  This also reduces the amount of checks we have to perform

Testing:

* Re-work 'inventory' generation by implementing a InventoryFactory.
* Added tests for c_v_commands

~depended on #1635 (hence https://github.com/SUSE/DeepSea/commit/611d6d98742811450f9d1f3de839a3929a993ef5)~ merged

-----------------

**Checklist:**
- [x] Added unittests and or functional tests
~- [ ] Adapted documentation~
- [x] Referenced issues or internal bugtracker
- [ ] Ran integration tests successfully (trigger with "@susebot run teuthology" in a GitHub comment; see the [wiki](https://github.com/SUSE/DeepSea/wiki/Testing) for more information)

### CODE REVIEW

**Comment by jschmid1** [comment]
@susebot run teuthology

**Comment by jschmid1** [comment]
One thing I discovered:

Let's say we have a cluster with 7 disks (5HDDs and 2SSDs). We create a drive_group like this
``` yaml
<snip>
data_devices:
  rotational: 1
  limit: 4
wal_devices:
  rotational: 0
  limit: 2
</snip>
```
On the first iteration, we get a proper setup with 4 OSDs backed by 2 WALs.

On the next iteration DeepSea will create a standalone OSD for the last (7th) disk that wasn't used since limit was in place.

I don't think we should adapt the logic as it's technically a flaw in the users drive_group specification. If there is the request to specifically exclude certain disks from the deployment, we have to find a solution for it.. 

1) Add a 'exclude' flag in drive_groups
2) Add a mechanism in Deepsea to mark ignored disks (`salt-run disks.ignore $disk`)
3) Add a mechanism to ceph-volume to mark ignored disks (`ceph-volume ignore $disk`)

ideas?

**Comment by tserong** [comment]
> ideas?

I'd wait for someone to make the request :-) but if we have to do this, I think I prefer adding an exclude flag in drive_groups, rather than something deeper down the stack. That way all the disk configuration is still in one place.

**Comment by jschmid1** [comment]
@susebot run teuthology

File: srv/salt/_modules/dg.py [hunk 1]
```diff
@@ -845,7 +900,11 @@ def deploy(**kwargs):
                 "Will not deploy using Drive-Groups."
                 "Please consult <insert doc/man> for guidance"
                 "on how to migrate to Drive-Groups")
-    return __salt__['helper.run'](c_v_commands(**kwargs))
+    c_v_command_list = c_v_commands(**kwargs)
```
Comment (jschmid1) [inline srv/salt/_modules/dg.py]: add log.debug statements

File: srv/salt/_modules/dg.py [hunk 2]
```diff
@@ -845,7 +900,11 @@ def deploy(**kwargs):
...
+    rets = []
+    for cmd in c_v_command_list:
```
Comment (jschmid1) [inline srv/salt/_modules/dg.py]: @susebot run teuthology

**Comment by jschmid1** [comment]
@susebot run teuthology

**Comment by jschmid1** [comment]
failing tests are unrelated to this PR

**Pull request merged**